### PR TITLE
Remove Scala 2.11 from crossScalaVersions

### PR DESCRIPTION
**Pull request by patriknw**
* because +buildRelease doesn't work with mixed crossScalaVersions
* let's use akka.build.scalaVersion when building for 2.11
* aggregatedProjects adjusted depending on akka.build.scalaVersion is 2.11 or not,
  there excluding/including the -typed modules

Refs https://github.com/akka/akka/issues/26651

Some Jenkins/Travis jobs will have to change and use `akka.build.scalaVersion`.

### CODE REVIEW

File: project/scripts/release [hunk 1]
```diff
@@ -299,6 +299,7 @@ else
   RELEASE_OPT="-Dakka.genjavadoc.enabled=true"
 fi
 try sbt $RELEASE_OPT +buildRelease
+try sbt -Dakka.build.scalaVersion=2.11.12 -D$RELEASE_OPT buildRelease
```
Comment (patriknw) [inline project/scripts/release]: one drawback here is that the pgp passphrase must be entered again, but I guess we can live with that (this will be removed for 2.6)

**Comment by patriknw** [comment]
I have verified this with a release dry-run. Checked
* api
* japi
* docs
* artifacts built for 2.11, 2.12 and 2.13, no 2.11 for -typed
* akka-streams jar contains jdk11 JavaFlowSupport classes

**Comment by patriknw** [comment]
updated travis build

File: project/scripts/release (hunk 1 above)
Comment (johanandren) [reply project/scripts/release]: I think that is ok for now

**Review by johanandren — APPROVED** [review]
LGTM,

there is a nightly 2.11 jenkins job that I changed recently will need changing when this has been merged.

File: build.sbt [hunk 2]
```diff
@@ -22,20 +22,21 @@ initialize := {
...
-// akka.AkkaBuild.buildSettings
+akka.AkkaBuild.buildSettings
 shellPrompt := { s => Project.extract(s).currentProject.id + " > " }
 resolverSettings
 
+def isScala211: Boolean = System.getProperty("akka.build.scalaVersion", "").startsWith("2.11")
+
 // When this is updated the set of modules in ActorSystem.allModules should also be updated
-lazy val aggregatedProjects: Seq[ProjectReference] = Seq(
+lazy val aggregatedProjects: Seq[ProjectReference] = List[ProjectReference](
```
Comment (helena) [inline build.sbt]: Just curious why the change to `List` explicitly

**Review by helena — APPROVED** [review]
LGTM

File: build.sbt (hunk 2 above)
Comment (patriknw) [reply build.sbt]: There is some magic implicits between Project and ProjectReference and it didn't like when I broke it up in two lists. Maybe it could still be Seq with the additional type annotations, but this works.

**Comment by patriknw** [comment]
Updated Jenkins job akka-nightly-2.11

**Comment by Falmarri** [comment]
Is this intentionally removing support for 2.11 from akka-typed-* ? Why now and not in akka 2.6?

**Comment by raboof** [comment]
Yes - there was a type inference issue in 2.11 that was fixed in 2.12 but not backported, so to make type inference work for akka-typed on 2.12 and 2.13 we had to drop 2.11 for those modules.
